src/packages/auth/functions.ts

```diff
@@ -96,11 +141,11 @@ export async function login(username: string, password: string): Promise<AuthRes
         };
       } else {
         const { ssoCookieValue: camToken } = json;
-        // TODO: Use other roles instead of 'admin'.
+        const { allowed_roles, default_role } = await getUserRoles(username);
         return {
           message: 'Login successful',
           success: true,
-          token: generateJwt(username, camToken, 'admin', ['admin', 'user']),
+          token: generateJwt(username, camToken, default_role, allowed_roles),
```

Comment:
The Auth Disabled branch needs to have the same changes.

Reply:
With auth disabled there is no call to CAM and thus "username" can be anything, and an admin role is implied. Do we really need to populate the user table in this case?

Reply:
Yes, as CAM is only in charge of _authenticating_ that a user is who they say they are as they log in. The database is in charge of _authorizing_ the user to use Aerie once they're logged in.

Reply:
So I don't think I phrased my question well. The JWT here gets an admin role, and that token authorizes the user to make any unrestricted query they want. Why is the users table needed if the JWT fully authorizes us with auth disabled? What else do we get from it?

Reply:
The Merlin DB has FKs set up with the Users Table. If a user with an `admin` role performs an interaction that would put their user-id in one of these columns (ie, creating a merge request), the interaction would fail due to the FK violation. Therefore, we need to insert into the `users` and `users_allowed_role` tables.

Reply:
Okay thanks - that makes sense

Reply:
Updated to include the roles from the DB for the no auth case